src/lib/leaflet-utils/layer-factory.js

```diff
@@ -11,7 +11,19 @@ import { greyIcon, redIcon, defaultIcon, iconsByLayerType } from '@/lib/leaflet-
 
 import './cluster-icon.css'
 
-export default async function createLayer (layer, { onClick }) {
+let vnode
+
+function emit (vnode, name, data) {
+  var handlers = (vnode.data && vnode.data.on) ||
+    (vnode.componentOptions && vnode.componentOptions.listeners)
+
+  if (handlers && handlers[name]) {
+    handlers[name].fns(data)
+  }
+}
+
+export default async function createLayer (layer, { onClick }, _, vnodeRef) {
+  vnode = vnodeRef
```

Comment:
can you explain what's going on with this vnode/vnodeRef?

Reply:
I added comments in the code to clarify on what's going on in my commit after this comment.